lib/config/validation.js

```diff
@@ -79,6 +79,23 @@ function validateConfig(config) {
             message: `Invalid semver range for allowedVersions: \`${val}\``,
           });
         }
+      } else if(key === 'packageRules' && val[0] !== undefined) {
+        const selectorsToValidate = ['packageNames', 'packagePatterns', 'excludePackageNames', 'excludePackagePatterns'];
+        let checkPackage = false;
+
+        for(const prop in val[0]){
+          if(selectorsToValidate.indexOf(prop) >= 0 && val[0][prop].length > 0){
+            checkPackage = true;
+            break;
+          }
+        }
```

Comment:
We still need to validate the fields too, e.g. if they have a selector then they must also have valid other fields. so the logic for validating an array must be added here otherwise it would get skipped